lightning/src/ln/channelmanager.rs

```diff
@@ -7206,6 +7344,33 @@ where
 	}
 }
 
+impl Writeable for VecDeque<(Event, Option<EventCompletionAction>)> {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		(self.len() as u64).write(w)?;
+		for elem in self.iter() {
+			elem.write(w)?;
+		}
+		Ok(())
+	}
+}
+impl Readable for VecDeque<(Event, Option<EventCompletionAction>)> {
+	fn read<R: Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let len: u64 = Readable::read(reader)?;
+		const MAX_ALLOC_SIZE: u64 = 1024 * 16;
+		let mut events = VecDeque::with_capacity(cmp::min(
+			MAX_ALLOC_SIZE/mem::size_of::<(events::Event, Option<EventCompletionAction>)>() as u64,
+			len) as usize);
+		for _ in 0..len {
+			let ev_opt = MaybeReadable::read(reader)?;
```

Comment:
If we fail to read an event and `action.is_some()`, seems we'll never release the update? Does that mean `PaymentSent`s should never break backwards compat?